scripts/histmakers/w_z_gen_dists.py

```diff
@@ -0,0 +1,104 @@
+import argparse
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--nThreads", type=int, help="number of threads", default=None)
+parser.add_argument("--maxFiles", type=int, help="Max number of files (per dataset)", default=-1)
+parser.add_argument("--filterProcs", type=str, nargs="*", help="Only run over processes matched by (subset) of name", default=["Wplus", "Wminus", "Zmumu", "Ztautau"])
+args = parser.parse_args()
+
+import ROOT
+ROOT.gInterpreter.ProcessLine(".O3")
+if not args.nThreads:
+    ROOT.ROOT.EnableImplicitMT()
+elif args.nThreads != 1:
+    ROOT.ROOT.EnableImplicitMT(args.nThreads)
+
+import pickle
+import gzip
+
+import narf
+import wremnants
+import hist
+import lz4.frame
+import logging
+
+filt = lambda x,filts=args.filterProcs: any([f in x.name for f in filts])
+datasets = wremnants.datasets2016.getDatasets(maxFiles=args.maxFiles, filt=filt if args.filterProcs else None)
```

Comment:
I can't quite tell where you are filtering this down to just W and Z, but you might as well do it here from the start

Reply:
I put it in the default filter arguments.  Open to better suggestions.

Reply:
Aaah got it. Is there a way to do this that limits the options allowed to be a subset of this? Maybe

nargs="+", choices=["Wplus", "Wminus", "Zmumu", "Ztautau"], default=["Wplus", "Wminus", "Zmumu", "Ztautau"]) ?

Reply:
I think we need to think of a better way of handling process groups/physics attributes in general to improve this further, so I would leave it like this for now.  (restricting the choices for wildcards like this doesn't make a lot of sense, since I could legitimately pass "WplusmunuPostVFP" as a filter)

If the user passes extra processes it'll either crash because of missing branches, or the extra procs won't be used for the angular coeffs at least because the final sums check explicitly ```wprocs``` and ```zprocs```